### PR TITLE
openshift: record etcd logs in test runs

### PR DESCRIPTION
**Pull request by jim-minter**

### CODE REVIEW

**Comment by jim-minter** [comment]
@kargakis @pweil- ptal

File: test/e2e/runner/cli_provisioner.go [hunk 1]
```diff
@@ -328,3 +329,34 @@ func (cli *CLIProvisioner) FetchOpenShiftInfraLogs(logPath string) error {
...
+func (cli *CLIProvisioner) FetchOpenShiftMachineLogs(cfg *config.Config, eng *engine.Engine, logPath string) error {
+	sshKeyPath := cfg.GetSSHKeyPath()
+
+	kubeConfig, err := kubernetes.GetConfig()
+	if err != nil {
+		return err
+	}
+	master := fmt.Sprintf("%s@%s", eng.ClusterDefinition.Properties.LinuxProfile.AdminUsername, kubeConfig.GetServerName())
+
+	services := []string{"etcd.service"}
```
Comment (kargakis) [inline test/e2e/runner/cli_provisioner.go]: Soon this will run as a pod. Can you try to separate the logic based on the orchestrator version?

Comment (jim-minter) [reply test/e2e/runner/cli_provisioner.go]: Follow-up.

**Comment by jim-minter** [comment]
@CecileRobertMichon please merge

**Comment by kargakis** [comment]
/lgtm

We should retrieve logs from the rest of the control plane in a follow-up. I am also looking into deploying all of our components with increased log verbosity as part of the e2e test deployment.

File: test/e2e/kubernetes/config.go [hunk 2]
```diff
@@ -45,5 +45,9 @@ func GetConfig() (*Config, error) {
 // GetServerName returns the server for the given config in an sshable format
 func (c *Config) GetServerName() string {
 	s := c.Clusters[0].ClusterInfo.Server
-	return strings.Split(s, "://")[1]
+	s = strings.Split(s, "://")[1]
+	if strings.ContainsRune(s, ':') {
```
Comment (pweil-) [inline test/e2e/kubernetes/config.go]: should this not be a parsable URL?

**Comment by jim-minter** [comment]
> We should retrieve logs from the rest of the control plane in a follow-up. I am also looking into deploying all of our components with increased log verbosity as part of the e2e test deployment.

@kargakis unclear if you want me to open an issue or if you're working on it already?

**Comment by kargakis** [comment]
> @kargakis unclear if you want me to open an issue or if you're working on it already?

Working on bumping the loglevel now, once this PR merges I will work on getting logs from the rest of the services.

File: test/e2e/kubernetes/config.go (hunk 2 above)
Comment (jim-minter) [reply test/e2e/kubernetes/config.go]: updated...

**Comment by jim-minter** [comment]
@CecileRobertMichon please merge

**Comment by kargakis** [comment]
```
$ kubectl config view -o json
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/Azure/acs-engine/test/e2e/kubernetes.(*Config).GetServerName(0xc4203400c0, 0x0, 0x0)
	/go/src/github.com/Azure/acs-engine/test/e2e/kubernetes/config.go:48 +0x1af
github.com/Azure/acs-engine/test/e2e/runner.(*CLIProvisioner).FetchOpenShiftMachineLogs(0xc4200ec750, 0xc420197720, 0xc420341140, 0xc420300000, 0x5a, 0xc4202ba060, 0x30)
	/go/src/github.com/Azure/acs-engine/test/e2e/runner/cli_provisioner.go:341 +0x77
main.teardown()
	/go/src/github.com/Azure/acs-engine/test/e2e/runner.go:204 +0x717
main.main()
	/go/src/github.com/Azure/acs-engine/test/e2e/runner.go:118 +0x6df
exit status 2
test.mk:32: recipe for target 'test-openshift' failed
```

**Comment by kargakis** [comment]
lgtm, @CecileRobertMichon please merge

**Comment by kargakis** [comment]
@acs-bot I keep forgetting

@CecileRobertMichon ping

**Comment by CecileRobertMichon** [comment]
@jim-minter @kargakis I restarted the openshift e2e twice and have not been able to get a single successful.I would like to get at least one e2e run since this PR is changing the test itself. It might be worth considering disabling the test until it is more reliable since there is little in a test that is red more often than it is green what do you think?

**Comment by jim-minter** [comment]
Hey @CecileRobertMichon we're also frustrated by the testing situation.  Two issues are that we're spread thin right now, and that we also don't have all the gating we need in place upstream on the origin and openshift-ansible repos so we're perpetually chasing breaks there.  My personal preference is to leave the tests here enabled and red rather than disabled - we will get to the flakes and get them green.

**Pull request merged**